usaspending_api/disaster/v2/views/agency/count.py

```diff
@@ -17,18 +19,28 @@ class AgencyCountViewSet(CountBase):
 
     @cache_response()
     def post(self, request: Request) -> Response:
-        filters = [
-            Q(treasury_account__funding_toptier_agency=OuterRef("pk")),
-            self.is_in_provided_def_codes(),
-            self.all_closed_defc_submissions,
-            self.is_non_zero_object_class_cpe(),
-        ]
-        count = (
-            ToptierAgency.objects.annotate(
-                include=Exists(FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters).values("pk"))
+        if self.award_type_codes:
+            filters = [
+                self.is_in_provided_def_codes(),
+                self.all_closed_defc_submissions,
+                self.is_non_zero_award_cpe(),
+            ]
+            count = FinancialAccountsByAwards.objects.filter(*filters)
+            count = count.values("award_id").filter(award__type__in=self.filters.get("award_type_codes"))
+            count = count.aggregate(count=Count("award__funding_agency__toptier_agency", distinct=True))["count"]
```

Comment:
stylistically I think the chaining of filters with just one return reads a bit better. May not be obvious to all readers that the queryset isn't evaluated til an action is taken on it.

e.g. 
```
count = (
    Model.objects.filter(...)
     .this(...)
     .that(...)
     .action(...)
)
```

You do that below, so it'd be good for consistency .